src/gui/screen_printing_serial.cpp

```diff
@@ -0,0 +1,162 @@
+#include "dbg.h"
+#include "gui.h"
+#include "config.h"
+#include "window_header.h"
+#include "status_footer.h"
+#include "marlin_client.h"
+#include "filament.h"
+#include "marlin_server.h"
+#include "guitypes.h"      //font_meas_text
+#include "stm32f4xx_hal.h" //HAL_GetTick
+
+//uncomment to use tune button
+//#define BUTTON_TUNE 1
+#ifdef BUTTON_TUNE
+extern screen_t *pscreen_menu_tune;
+#endif //BUTTON_TUNE
+
+#pragma pack(push)
+#pragma pack(1)
+
+typedef struct
+{
+    window_frame_t root;
+
+    window_header_t header;
+    status_footer_t footer;
+
+    window_text_t w_message; //Messages from onStatusChanged()
+    window_icon_t octo_icon;
+#ifdef BUTTON_TUNE
+    window_icon_t tune_button;
+    window_text_t tune_label;
+#endif //BUTTON_TUNE
+    int last_tick;
+
+} screen_printing_serial_data_t;
+
+#pragma pack(pop)
+
+void screen_printing_serial_init(screen_t *screen);
+void screen_printing_serial_done(screen_t *screen);
+void screen_printing_serial_draw(screen_t *screen);
+int screen_printing_serial_event(screen_t *screen, window_t *window, uint8_t event, void *param);
+
+static const char txt0[] = "Remote printing\nin progress    ";
+static const char txt1[] = "Remote printing\nin progress .  ";
+static const char txt2[] = "Remote printing\nin progress .. ";
+static const char txt3[] = "Remote printing\nin progress ...";
```

Comment:
this is brutal ... why don't you append the dots to the text when being printed?

Reply:
I did try to redraw just dots. But there is an bug in textfield.
Redrawing dots erased parts of characters around it.